packages/cli/src/util/autoVersion.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2023 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { execSync } from "node:child_process";
+import { isValidSemver } from "./isValidSemver.js";
+
+/**
+ * Gets the version string using git describe.
```

Comment:
Worth a note about the behaviour when tagPrefix is empty string? (all tags, expects version as tag possibly with v prefix) We'll have to document this on the other layers on the CLI and user docs as well later obviously